src/plugins/workspace/server/workspace_client.ts

```diff
@@ -89,6 +104,7 @@ export class WorkspaceClient implements IWorkspaceClientImpl {
     payload: Omit<WorkspaceAttributeWithPermission, 'id'>
   ): ReturnType<IWorkspaceClientImpl['create']> {
     try {
+      this.hasOSDAdminPermission(requestDetail);
```

Comment:
Having this check in workspace client may not enough, as people may call saved object api to create workspace directly. Maybe better to have it in the saved object client wrapper. What do you think? cc @SuZhou-Joe

Reply:
Agree, let's do it in the permission wrapper so that we can ensure the check will be performed only when permission control feature flag is enabled.

Reply:
Thanks, I will move the related logic to saved object client.